Assets/Scripts/StandardSamples/Services/EOSService.cs

```diff
@@ -23,79 +23,88 @@
+        public delegate void ServiceUpdatedEventHandler();
 
         /// <summary>
-        /// Indicates whether the service needs to have a user authenticated in
-        /// order to function properly.
+        /// Event that triggers when changes have been made that may require a
+        /// user interface update.
         /// </summary>
-        private bool _requiresLoggedInWithConnectInterface;
+        public event ServiceUpdatedEventHandler Updated;
 
         /// <summary>
-        /// Called when connect login has taken place.
+        /// Indicates whether the service needs to have a user authenticated in
+        /// order to function properly.
         /// </summary>
-        /// <param name="loginCallbackInfo"></param>
-        public void OnConnectLogin(LoginCallbackInfo loginCallbackInfo)
-        {
-            if (loginCallbackInfo.ResultCode != Result.Success)
-            {
-                UnityEngine.Debug.Log($"Player login did not succeed. Result code: {Enum.GetName(typeof(Result), loginCallbackInfo.ResultCode)}");
-                return;
-            }
-            OnPlayerLogin(loginCallbackInfo.LocalUserId);
-        }
+        protected bool RequiresAuthentication { get; }
 
         /// <summary>
         /// Base constructor for Service managers.
         /// </summary>
-        /// <param name="requiresLoggedInWithConnectInterface">
+        /// <param name="requiresAuthentication">
         /// Indicates whether the service manager requires a user to be
         /// authenticated with the Connect Interface in order to function
         /// properly.
         /// </param>
-        protected EOSService(bool requiresLoggedInWithConnectInterface)
+        protected EOSService(bool requiresAuthentication = true)
         {
-            _requiresLoggedInWithConnectInterface = requiresLoggedInWithConnectInterface;
-
-            EOSManager.Instance.AddConnectLoginListener(this);
-
+            RequiresAuthentication = requiresAuthentication;
+            AuthenticationListener.Instance.AuthenticationChanged += OnAuthenticationChanged;
             _ = RefreshAsync();
         }
 
+        ~EOSService()
+        {
+            Dispose();
+        }
+
         /// <summary>
-        /// Determines whether the user is currently authenticated using the
-        /// Connect Interface provided by the EOS SDK.
+        /// Try to get the product user id from the connect interface.
         /// </summary>
+        /// <param name="productUserId">
+        /// ProductUserId for the currently logged in user.
+        /// </param>
         /// <returns>
-        /// True if the user is authenticated using the Connect Interface, false
-        /// otherwise.
+        /// True if the ProductUserId was successfully retrieved and is valid,
+        /// false otherwise.
         /// </returns>
-        private static bool IsLoggedInWithConnectInterface()
+        protected bool TryGetProductUserId(out ProductUserId productUserId)
```

Comment:
I take this will mean a service should utilize this function, rather than fetching from `EOSManager.Instance.GetProductUserId()`?

Reply:
I added this because in the process of refactoring things, I had more than one class that derived from `EOSService` doing the same thing. It's up to the individual developer to decide whether or not to utilize this, but this helper function seemed like a clean way to simplify things without changing the API (since changes to `EOSManager` constitute changing the API).